src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -33,43 +41,72 @@ export class PtyService extends Disposable implements IPtyService {
 	readonly onProcessOverrideDimensions = this._onProcessOverrideDimensions.event;
 	private readonly _onProcessResolvedShellLaunchConfig = this._register(new Emitter<{ id: number, event: IShellLaunchConfig }>());
 	readonly onProcessResolvedShellLaunchConfig = this._onProcessResolvedShellLaunchConfig.event;
-
 	constructor(
 		private readonly _logService: ILogService
 	) {
 		super();
 	}
 
+	onPtyHostExit?: Event<number> | undefined;
+	onPtyHostStart?: Event<void> | undefined;
+
+	orphanQuestionReply(args: IOrphanQuestionReplyArgs): Promise<void> {
+		return this._throwIfNoPty(args.id).orphanQuestionReply();
+	}
+
+	async acknowledgeDataEvent(id: number, charCount: number): Promise<void> {
+		return this._throwIfNoPty(id).acknowledgeCharCount(charCount);
+	}
+
+	// following suit with the remoteTerminalService impl
+	async getLatency(id: number): Promise<number> {
+		return 0;
+	}
+
 	dispose() {
 		for (const pty of this._ptys.values()) {
 			pty.shutdown(true);
 		}
 		this._ptys.clear();
 	}
 
-	async createProcess(shellLaunchConfig: IShellLaunchConfig, cwd: string, cols: number, rows: number, env: IProcessEnvironment, executableEnv: IProcessEnvironment, windowsEnableConpty: boolean): Promise<number> {
+	async fetchPersistentTerminalProcess(id: number): Promise<number> {
```

Comment:
Let's call this `attachToProcess` to align with `createProcess` (which is also a persistent process)